### PR TITLE
wholesale: add support for routing tags

### PR DESCRIPTION
**Pull request by Kaian**
Added a basic implementation of routing tags that allows wholesale to prefix their calls with special tags to select the outgoing routing that will be used..

This implements #489 and require #500 implement to handle corner case updates of Outgoing routings

### CODE REVIEW

**Comment by Kaian** [comment]
You can't breath in this project without breaking a couple of tests

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/RoutingPattern.RoutingPattern.orm.yml [hunk 1]
```diff
@@ -12,6 +12,12 @@ Ivoz\Provider\Domain\Model\RoutingPattern\RoutingPattern:
       generator:
         strategy: IDENTITY
   oneToMany:
+    outgointRoutings:
```
Comment (cruzccl) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/Mapping/RoutingPattern.RoutingPattern.orm.yml]: typo, man!

**Comment by cruzccl** [comment]
Blocked by #502

**Comment by Kaian** [comment]
Updated with requested changes.

File: kamailio/trunks/config/kamailio.cfg [hunk 2]
```diff
@@ -520,6 +526,9 @@ route[SELECT_GW] {
         exit;
     }
 
+    xerr("Flags: $avp(peerServerId)\n");
+    xerr("Rule id: $avp(ruleId)\n");
```
Comment (Kaian) [inline kamailio/trunks/config/kamailio.cfg]: Are these debugging lines? :thinking:

File: kamailio/users/config/kamailio.cfg [hunk 3]
```diff
@@ -561,6 +562,38 @@ route[ADAPT_RURI_IN] {
     $rU = $var(transformated);
 }
 
+# Detects routing tag, sets header with tag and strips it from r-uri
+route[GET_ROUTING_TAG] {
+    if ($dlg_var(type) == 'wholesale') return;
+
+    # Guess routing tag
+    sql_query("cb", "SELECT tag FROM RoutingTags RT INNER JOIN CompaniesRelRoutingTags CRRT ON CRRT.routingTagId = RT.id WHERE CRRT.companyId = '$dlg_var(companyId)'", "rb");
```
Comment (Kaian) [inline kamailio/users/config/kamailio.cfg]: prefix column for RoutingTags is called `prefix` unless you're suggesting renaming to `tag` to match this query

Comment (cruzccl) [reply kamailio/users/config/kamailio.cfg]: Sorry, I suggested the renaming to the mate on the right but I forgot telling you...

Comment (Kaian) [reply kamailio/users/config/kamailio.cfg]: Don't bother with that guy, he's as handsome as forgetful
